mlflow/spark.py

```diff
@@ -235,6 +239,9 @@ def log_model(
     try:
         spark_model.save(posixpath.join(model_dir, _SPARK_MODEL_PATH_SUB))
     except Py4JError:
+        if databricks_utils.is_in_cluster() and databricks_utils.is_mlflowdbfs_available():
```

Comment:
Will MLflowDBFS ever be available outside of Cluster? That is, can we move `databricks_utils.is_in_cluster()` inside `databricks_utils.is_mlflowdbfs_available()`?